tests/test_modifiers.py

```diff
@@ -174,3 +166,67 @@ def test_shapesys_holes():
         True,
         False,
     ]
+
+
+def test_invalid_bin_wise_modifier():
+    """
+    Test that bin-wise modifiers will raise an exception if their data shape
+    differs from their sample's.
+    """
+    spec = {
+        "channels": [
+            {
+                "name": "channel_1",
+                "samples": [
+                    {
+                        "name": "sample_1",
+                        "data": [1, 2, 3, 4],
+                        "modifiers": [
+                            {"name": "mu", "type": "normfactor", "data": None},
+                        ],
+                    },
+                    {
+                        "name": "sample_2",
+                        "data": [2, 4, 6, 8],
+                        "modifiers": [],
+                    },
+                ],
+            }
+        ],
+    }
+
+    assert pyhf.Model(spec)
+
+    bad_histosys_modifier = [
+        {
+            "name": "histosys_bad",
+            "type": "histosys",
+            "data": {
+                "hi_data": [3, 6, 9],
+                "lo_data": [1, 2, 3],
+            },
+        }
+    ]
+    bad_shapesys_modifier = [
+        {
+            "name": "shapesys_bad",
+            "type": "shapesys",
+            "data": [1, 2, 3],
+        }
+    ]
+    bad_staterror_modifier = [
+        {
+            "name": "staterror_bad",
+            "type": "staterror",
+            "data": [1, 2, 3],
+        }
+    ]
```

Comment:
If people have recommendations on how to parameterize this test I'm happy to make those changes.

Reply:
make three different workspaces, store as JSON files, and load them up -- and parametrize the paths?

Reply:
I did this at first, but then it seemed liked I had gone through more work and made it less clear what was actually being tested in each of the 3 situations and so went to this. Thoughts?

Reply:
hard-coding the spec I think is a better way to go.

Reply:
But it's probably not as clear.. hrm..

Reply:
I'll approve for now and unresolve if I convince myself otherwise.

Reply:
> hard-coding the spec I think is a better way to go.

Okay we can change it then. Or we could make these patch files and patch the spec JSON